smartsim/_core/mli/infrastructure/storage/featurestore.py

```diff
@@ -27,6 +27,21 @@
 import typing as t
 from abc import ABC, abstractmethod
 
+from pydantic import BaseModel, Field
```

Comment:
Per the group discussion, do we want to change this to a dataclass or do that in another pass?

Reply:
I'd like to knock out the replacement in a [new ticket](https://github.com/CrayLabs/Poseidon/issues/753)